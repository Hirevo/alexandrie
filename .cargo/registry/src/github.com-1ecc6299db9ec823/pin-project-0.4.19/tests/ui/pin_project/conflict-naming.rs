use pin_project::pin_project;

#[pin_project(project = A, project_ref = A)] //~ ERROR E0428,E0308
struct Struct(#[pin] ());

fn main() {}
