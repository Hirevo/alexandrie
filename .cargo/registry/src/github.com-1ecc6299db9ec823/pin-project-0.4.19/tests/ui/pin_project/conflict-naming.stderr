error[E0428]: the name `A` is defined multiple times
 --> $DIR/conflict-naming.rs:3:1
  |
3 | #[pin_project(project = A, project_ref = A)] //~ ERROR E0428,E0308
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | `A` redefined here
  | previous definition of the type `A` here
  |
  = note: `A` must be defined only once in the type namespace of this module
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
 --> $DIR/conflict-naming.rs:3:1
  |
3 | #[pin_project(project = A, project_ref = A)] //~ ERROR E0428,E0308
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ types differ in mutability
  |
  = note: expected mutable reference `&mut ()`
                     found reference `&()`
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
